sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/ArgumentMatcher.kt

```diff
@@ -33,29 +34,30 @@ class ArgumentMatcher(
     private val typeName: String? = null,
     private val nullability: TypeNullability? = null,
     private val qualified: Boolean = true,
+    val isVararg: Boolean = false,
 ) {
     companion object {
         val ANY = ArgumentMatcher()
     }
 
-    fun matches(descriptor: ValueParameterDescriptor) =
-        matchesName(descriptor) && matchesNullability(descriptor)
+    fun matches(descriptor: ValueParameterDescriptor) = (isVararg == (descriptor.varargElementType != null)) &&
+        matchesName(if (isVararg) descriptor.varargElementType else descriptor.type) && matchesNullability(descriptor)
```

Comment:
Minor detail: I just noticed we are doing the nullability check after the type check. For performance reasons, we could switch them.